api/src/main/java/org/opennms/integration/api/v1/ui/UIExtension.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.integration.api.v1.ui;
+
+/**
+ * The interface to implement UI extension plugin to be installed as OSGI feature in OpenNMS
+ */
+public interface UIExtension {
+    /**
+     * @return the unique UI extension ID
+     */
+    String getExtensionID();
```

Comment:
For consistency:
```suggestion
    String getExtensionId();
```

See https://github.com/OpenNMS/opennms-integration-api/blob/a169a904de33e40f417b08e09993f1f430696d27/api/src/main/java/org/opennms/integration/api/v1/dao/NodeDao.java#L54